5_analyze_data/notebooks/UMAP_analysis/UMAPutils.py

```diff
@@ -0,0 +1,68 @@
+import pathlib
+import pandas as pd
+
+def split_data(pycytominer_output: pd.DataFrame):
+    """
+    split pycytominer output to metadata dataframe and np array of feature values
+    Parameters
+    ----------
+    pycytominer_output : pd.DataFrame
+        dataframe with pycytominer output
+    Returns
+    -------
+    pd.Dataframe, np.ndarray
+        metadata dataframe, feature values
+    """
+    # split metadata from features
+    metadata_cols = [
+        col_name
+        for col_name in pycytominer_output.columns.tolist()
+        if "Metadata" in col_name
+    ]
+    metadata_dataframe = pycytominer_output[metadata_cols]
+
+    feature_cols = [
+        col_name
+        for col_name in pycytominer_output.columns.tolist()
+        if "Metadata" not in col_name
```

Comment:
How would this work? 

@d33bs

Reply:
@jenna-tomkinson One way to accomplish this might be to take the complement of the columns within `metadata_cols` by using `pycytominer_output.columns.difference(metadata_cols)`. See [here](https://datascience.stackexchange.com/questions/46434/dataframe-columns-difference-use) for more details.

Reply:
Thank you for the clarification! I have removed the extraction of feature data since it is not used during the UMAP creation process.